userland/libtock/ble.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*******************************************************************************
+ *   DRIVER DEFINITIONS   -- corresponds to different system calls
+ *
+ *    *_CMD   - command call
+ *    *_ALLOW - allow call
+ *    *_SUB   - subscribe call
+ *
+ ******************************************************************************/
+
+#define BLE_DRIVER_NUMBER 0x30000
+
+#define BLE_ADV_START_CMD 0
+#define BLE_ADV_STOP_CMD 1
+#define BLE_CFG_TX_POWER_CMD 2
+#define BLE_SCAN_CMD 5
+
+#define BLE_SCAN_SUB 0
+
+#define BLE_CFG_ADV_BUF_ALLOW 0
+#define BLE_CFG_SCAN_BUF_ALLOW 1
+
+#define ADV_IND  0x00
+#define ADV_DIRECT_IND  0x01
+#define ADV_NON_CONN_IND  0x02
+#define ADV_SCAN_IND  0x06 
+
+typedef enum {
+  POSITIVE_10_DBM = 10,
+  POSITIVE_9_DBM = 9,
+  POSITIVE_8_DBM = 8,
+  POSITIVE_7_DBM = 7,
+  POSITIVE_6_DBM = 6,
+  POSITIVE_5_DBM = 5,
+  POSITIVE_4_DBM = 4,
+  POSITIVE_3_DBM = 3,
+  POSITIVE_2_DBM = 2,
+  POSITIVE_1_DBM = 1,
+  ZERO_DBM = 0,
+  NEGATIVE_1_DBM = 0xff,
+  NEGATIVE_2_DBM = 0xfe,
+  NEGATIVE_3_DBM = 0xfd,
+  NEGATIVE_4_DBM = 0xfc,
+  NEGATIVE_5_DBM = 0xfb,
+  NEGATIVE_6_DBM = 0xfa,
+  NEGATIVE_7_DBM = 0xf9,
+  NEGATIVE_8_DBM = 0xf8,
+  NEGATIVE_9_DBM = 0xf7,
+  NEGATIVE_10_DBM = 0xf6,
+  NEGATIVE_11_DBM = 0xf5,
+  NEGATIVE_12_DBM = 0xf4,
+  NEGATIVE_13_DBM = 0xf3,
+  NEGATIVE_14_DBM = 0xf2,
+  NEGATIVE_15_DBM = 0xf1,
+  NEGATIVE_16_DBM = 0xf0,
+  NEGATIVE_17_DBM = 0xef,
+  NEGATIVE_18_DBM = 0xee,
+  NEGATIVE_19_DBM = 0xed,
+  NEGATIVE_20_DBM = 0xec,
+} TxPower_t;
+
+// start advertising
+//
+// pdu_type           - Type of advertising PDU. One of ADV_IND, ADV_NON_CONN or ADV_SCAN_IND
+// advd               - The advertising data
+// len                - Length of the advertising data (will be truncated to 31 bytes)
+// interval           - The advertising interval in milliseonds
+int ble_start_advertising(int pdu_type, uint8_t* advd, int len, uint16_t interval);
```

Comment:
Would be a little bit more ergonomic to pass `AdvData` instead of `advd` and `len` here!

Reply:
Yes, exactly.

Reply:
Hahaha. I also had a comment before I saw this wanting to pass in `AdvData`. Could we have a `simple-ble` library call that accepts an `AdvData` and calls `ble_start_advertising`? Then ergonomically it would still be good for users. Or is it not possible to have other libraries depend on libtock?